tests/test_projectbuilder.py

```diff
@@ -60,35 +59,35 @@ def from_name(cls, name):
 
 
 @pytest.mark.parametrize(
-    ('requirement_string', 'expected', 'extra_warning'),
+    ('requirement_string', 'expected'),
     [
-        ('something', True, False),
-        ('something_else', False, False),
-        ('something[extra]', False, False),
-        ('something[some_extra]', True, True),
-        ('something_else; python_version>"10"', True, False),
-        ('something_else; python_version<="1"', True, False),
-        ('something_else; python_version>="1"', False, False),
-        ('something == 1.0.0', True, False),
-        ('something == 2.0.0', False, False),
-        ('something[some_extra] == 1.0.0', True, True),
-        ('something[some_extra] == 2.0.0', False, True),
+        ('something', None),
+        ('something_else', ('something_else',)),
+        # Undefined extra
+        ('something[extra]', None),
+        # Extra without associated dependencies (would the wheel builder filter this out?)
+        ('something[some_extra]', None),
+        # Extra with missing dependencies
+        (
+            'something[some_other_extra]',
+            ('something[some_other_extra]', "some_other_extra_dependency; extra == 'some_other_extra'"),
+        ),
+        # Extra with its dependencies satisfied
+        ('something[the_quintessential_extra]', None),
+        ('something_else; python_version>"10"', None),
+        ('something_else; python_version<="1"', None),
+        ('something_else; python_version>="1"', ('something_else; python_version>="1"',)),
+        ('something == 1.0.0', None),
+        ('something == 2.0.0', ('something == 2.0.0',)),
+        ('something[some_extra] == 1.0.0', None),
+        ('something[some_extra] == 2.0.0', ('something[some_extra] == 2.0.0',)),
     ],
 )
-def test_check_version(requirement_string, expected, extra_warning):
-    if extra_warning:
-        with pytest.warns(build.IncompleteCheckWarning):
-            assert build.check_version(requirement_string) == expected
-    else:
-        assert build.check_version(requirement_string) == expected
+def test_check_dependency(requirement_string, expected):
```

Comment:
If we are gonna merge this here, we need to also test recursive extra dependencies.

Reply:
Added.